types/gensync/index.d.ts

```diff
@@ -0,0 +1,145 @@
+// Type definitions for gensync 1.0
+// Project: https://github.com/loganfsmyth/gensync
+// Definitions by: Jake Bailey <https://github.com/jakebailey>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.0
+
+/**
+ * Returns a function that can be "awaited" (with `yield*`) in another `gensync` generator
+ * function, or executed via
+ *
+ *   - `.sync(...args)` - Returns the computed value, or throws.
+ *   - `.async(...args)` - Returns a promise for the computed value.
+ *   - `.errback(...args, (err, result) => {})` - Calls the callback with the computed value, or error.
+ * @param generatorFnOrOptions A generator function, or options for an existing sync/async function
+ */
+declare function gensync<A extends unknown[], R, E = unknown>(
```

Comment:
I defaulted `E` to unknown everywhere; I'm not sure if this is good or if `any` would be a better choice. Hard to claw back from `any` once it's in the types, though, so I'm being strict.